application_form/tests/api/test_apartment_reservation_api.py

```diff
@@ -1000,3 +1001,58 @@ def test_salesperson_create_reservation_generate_metadata(
     reservation = ApartmentReservation.objects.get(id=reservation_id)
     user = sales_ui_salesperson_api_client.user
     assert reservation.handler == f"{user.first_name} {user.last_name}"
+
+
+@pytest.mark.django_db
+def test_release_pdf_creation_unauthorized(elasticsearch, user_api_client):
+    apartment = ApartmentDocumentFactory()
+    reservation = ApartmentReservationFactory(apartment_uuid=apartment.uuid)
+
+    response = user_api_client.get(
+        reverse(
+            "application_form:sales-apartment-reservation-release-pdf",
+            kwargs={"pk": reservation.id},
+        ),
+        format="json",
+    )
+
+    assert response.status_code == 403
+
+
+@pytest.mark.parametrize("reservation_has_revaluation", (True, False))
+@pytest.mark.parametrize("ownership_type", ("HASO", "Hitas"))
+@pytest.mark.django_db
+def test_release_pdf_creation(
+    elasticsearch,
+    sales_ui_salesperson_api_client,
+    ownership_type,
+    reservation_has_revaluation,
+):
+    apartment = ApartmentDocumentFactory(project_ownership_type=ownership_type)
+
+    reservation = ApartmentReservationFactory(
+        apartment_uuid=apartment.uuid, state=ApartmentReservationState.CANCELED
+    )
+    revaluation = None
+    if reservation_has_revaluation:
+        revaluation = ApartmentRevaluationFactory(apartment_reservation=reservation)
```

Comment:
`revaluation = ApartmentRevaluationFactory(apartment_reservation=reservation) if reservation_has_revaluation else None` would be more PYTHONIC I guess.